onnxruntime/core/providers/cuda/cuda_mempool_arena.h

```diff
@@ -0,0 +1,169 @@
+﻿// Copyright (c) Microsoft.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include <cuda_runtime_api.h>
+#include <cstdint>
+#include <mutex>
+
+#include "core/common/common.h"                      // ORT_DISALLOW_COPY_ASSIGNMENT_AND_MOVE, ORT_THROW/ENFORCE
+#include "core/common/inlined_containers.h"          // InlinedHashMap, InlinedHashSet, InlinedVector
+#include "core/providers/cuda/cuda_stream_handle.h"  // ORT Stream -> cudaStream_t
+#include "core/providers/shared_library/provider_api.h"
+
+namespace onnxruntime {
+namespace logging {
+class Logger;
+}
+namespace cuda {
+/**
+ * @brief Stream-aware CUDA allocator implemented on top of a private `cudaMemPool_t`.
+ *        The purpose of this arena is to assist with memory allocations in environments where
+ *        a single process is hosting more than one cuda session. This arena hosts cuda memory pool
+ *        which has some tunable parameters to control its memory usage and de-allocates memory back to
+ *        the device according to the specified params. This is opposite to the BFCArena which only
```

Comment:
Not understanding this comment: "This is opposite to the BFCArena...."

IIUC - this mempool also gets trimmed only during `Shrink()` which gets called only on `OnRunEnd()`. Wouldn't that behavior be exactly the same as the BFCArena's ?